python/lsst/ip/diffim/getTemplate.py

```diff
@@ -223,89 +232,202 @@ def run(self, coaddExposures, bbox, wcs, dataIds, physical_filter=None, **kwargs
-
-        # Combine images from individual patches together
-        statsFlags = afwMath.stringToStatisticsProperty('MEAN')
-        statsCtrl = afwMath.StatisticsControl()
-        statsCtrl.setNanSafe(True)
-        statsCtrl.setWeighted(True)
-        statsCtrl.setCalcErrorMosaicMode(True)
-
-        templateExposure = afwImage.ExposureF(finalBBox, finalWcs)
-        templateExposure.maskedImage.set(np.nan, afwImage.Mask.getPlaneBitMask("NO_DATA"), np.nan)
-        xy0 = templateExposure.getXY0()
-        # Do not mask any values
-        templateExposure.maskedImage = afwMath.statisticsStack(maskedImageList, statsFlags, statsCtrl,
-                                                               weightList, clipped=0, maskMap=[])
-        templateExposure.maskedImage.setXY0(xy0)
+        Returns
+        -------
+        images : `list` [`lsst.afw.image.MaskedImage`]
+            MaskedImages of each of the input exposures, for warping.
+        catalog : `lsst.afw.table.ExposureCatalog`
+            Catalog of metadata for each exposure
+        totalBox : `lsst.geom.Box2I`
+            The union of the bounding boxes of all the input exposures.
+        """
+        catalog = afwTable.ExposureCatalog(self.schema)
+        catalog.reserve(len(exposures))
+        images = [exposure.maskedImage for exposure in exposures]
+        totalBox = geom.Box2I()
+        for coadd, dataId in zip(exposures, dataIds):
+            totalBox = totalBox.expandedTo(coadd.getBBox())
+            record = catalog.addNew()
+            record.setPsf(coadd.psf)
+            record.setWcs(coadd.wcs)
+            record.setPhotoCalib(coadd.photoCalib)
+            record.setBBox(coadd.getBBox())
+            record.setValidPolygon(afwGeom.Polygon(geom.Box2D(coadd.getBBox()).getCorners()))
+            record.set("tract", dataId["tract"])
+            record.set("patch", dataId["patch"])
+            # TODO: Modify this once we know which coadds were actually included?
+            record.set("weight", 1)
+
+        return images, catalog, totalBox
+
+    def _merge(self, maskedImages, bbox, wcs):
+        """Merge the images that came from one tract into a one larger image.
+
+        Parameters
+        ----------
+        maskedImages : `list` [`lsst.afw.image.MaskedImage`]
+            Images to be merged into one larger bounding box.
+        bbox : `lsst.geom.Box2I`
+            Bounding box defining the image to merge into.
+        wcs : `lsst.afw.geom.SkyWcs`
+            WCS of all of the input images to set on the output image.
+
+        Returns
+        -------
+        merged : `lsst.afw.image.MaskedImage`
+            Merged image with all of the inputs at their respective bbox
+            positions.
+        """
+        merged = afwImage.ExposureF(bbox, wcs)
+        weights = afwImage.ImageF(bbox)
+        for maskedImage in maskedImages:
+            weight = afwImage.ImageF(maskedImage.variance.array**(-0.5))
```

Comment:
Should comment here or a little lower that we expect `weight` to be zero for the `is_nan` pixels automatically.  Might even be worth a check if it's not too expensive.

Reply:
Should I just make `is_nan` the union of the `NaN` image pixels and the non-finite variance pixels?

Reply:
Sure, that sounds reasonable.